tests/test_qubit_device.py

```diff
@@ -1371,3 +1373,79 @@ def test_batch_size_always_None(self, mock_qubit_device, shape):
             dev._get_batch_size(tensor1, broadcasted_shape, qml.math.prod(broadcasted_shape))
             is None
         )
+
+
+class TestResourcesTracker:
```

Comment:
non-blocking: do we need a test that calls Tracker on a circuit or gradient execution? e.g., 

```
with qml.Tracker(dev) as tracker:
    qml.grad(circuit)(x)
```

Reply:
I tested locally that this works. I can add a test as well 👍🏼

Reply:
Added 👍🏼